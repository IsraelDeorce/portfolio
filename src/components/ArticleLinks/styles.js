import styled from 'styled-components'

export const ArticleLinks = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 1em;
  & > * {
    color: ${({ theme }) => theme.c_brackets};
    &:hover {
      color: ${({ theme }) => theme.c_varType};
    }
    padding-bottom: 1em;
  }
`
