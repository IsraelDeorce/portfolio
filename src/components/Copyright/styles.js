import styled from 'styled-components'

export const Copyright = styled.div`
  display: flex;
  justify-content: center;
`
