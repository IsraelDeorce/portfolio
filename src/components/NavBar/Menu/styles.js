import styled, { css } from 'styled-components'

export const Menu = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  ${({ isOpen }) => isOpen && css`
    flex-direction: column-reverse;;
    justify-content: flex-end;
    align-items: flex-start;
    background-color: ${({ theme }) => theme.c_menu};
    height: 100vh;
    width: 75vw;
    position: absolute;
    bottom: 0;
    left: 0;
    top: 0;
    box-shadow: blue;
    z-index: 2;
    transition: transform 0.3s ease-in-out;
    box-shadow: 0 5px 30px ${({ theme }) => theme.c_tab};
    padding-left: 2em;
  `}
`

/*
opacity: 0;
-webkit-transform: translate3d(-100%,0,0);
transition-property: opacity,visibility,transform,-webkit-transform;
transition-timing-function: ease-in-out;
transform: translate3d(-100%,0,0);
transition-duration: .25s;
*/


export const UserPreferences = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (max-width: 992px) {
    display: none;
    ${({ isOpen }) => isOpen && css`
      display: flex;
      border-bottom: 2px solid gray;
      padding: 0.35em 0;
      margin-bottom: 0.35em;
    `};
  }
`

export const LanguageSelector = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`

export const LanguageCode = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`