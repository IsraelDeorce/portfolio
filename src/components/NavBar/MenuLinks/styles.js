import styled, { css } from 'styled-components'

export const MenuLinks = styled.div`
  display: none
  & > * {
    cursor: pointer;
  }
  & > *:hover {
    color: ${({ theme }) => theme.c_varType};
  }
  @media (min-width: 992px) {
    display: flex;
    & > *:not(:last-child) {
      margin-right: 2em;
    }
  }
  @media (max-width: 992px) {
    display: none;
    
    ${({ isMenuOpen }) => isMenuOpen && css`
      display: flex;
      flex-direction: column;
      & > * {
        margin: .75em 0;
      }
    `}
  }
`
