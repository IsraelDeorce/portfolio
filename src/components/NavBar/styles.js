import styled from 'styled-components'

export const NavBar = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
  @media (min-width: 992px) {
    display: none;
  }
`

export const Menu = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (max-width: 992px) {
    display: none;
  }
`
