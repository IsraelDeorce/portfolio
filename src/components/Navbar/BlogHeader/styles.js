import styled from 'styled-components'

export const BlogHeader = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (min-width: 992px) {
    display: none;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
`
