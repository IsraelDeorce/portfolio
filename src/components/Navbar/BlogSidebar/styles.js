import styled from 'styled-components'

export const Background = styled.div`
  opacity: 0.7;
  background-color: ${({ theme }) => theme.c_background};
  position:fixed;
  width:100%;
  height:100%;
  top:0px;
  left:0px;
  @media (min-width: 992px) {
    display: none;
  }
`

export const BlogSidebar = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.c_menu};
  height: 100vh;
  width: 65vw;
  z-index: 2;
  box-shadow: 0 5px 30px ${({ theme }) => theme.c_tab};
  padding: 0 1em 1em 1em;
  & > * {
    width: 100%;
  }
  & > :last-child {
    align-self: center;
  }
  @media (min-width: 992px) {
    position: static;
    top: auto;
    left: auto;
    right: auto;
    height: auto;
    width: 285px;
    min-width: 285px;
    box-shadow: none;
  }
`

export const Header = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  min-height: 60px;
  border-bottom: 2px solid gray;
  & > *:not(:last-child) {
    padding: 0.35em;
  }
`

export const Preferences = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
`
