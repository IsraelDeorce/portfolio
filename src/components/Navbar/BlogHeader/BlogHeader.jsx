import { useDispatch } from 'react-redux'
import { Link } from 'react-router-dom'
import { MdMenu } from 'react-icons/md'
import { TitledLogo } from '../../'
import * as S from './styles'

function BlogHeader() {
  const dispatch = useDispatch()

  const openBlogSidebar = () => dispatch.session.update({ isBlogSidebarOpen: true })

  return (
    <S.BlogHeader>
      <S.Hamburguer onClick={openBlogSidebar}>
        <MdMenu size='32' />
      </S.Hamburguer>
      <Link to='/'>
        <TitledLogo />
      </Link>
    </S.BlogHeader>
  )
}

export default BlogHeader
