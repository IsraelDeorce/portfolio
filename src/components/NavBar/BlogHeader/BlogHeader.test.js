import { useDispatch } from 'react-redux'
import BlogHeader from './BlogHeader'
import * as S from './styles'

describe('<BlogHeader/>', () => {
  const dispatch = useDispatch()

  afterEach(() => jest.clearAllMocks())

  it('calls the open side bar function so update the store with side bar is open', () => {
    shallow(<BlogHeader />).find(S.Hamburguer).simulate('click')
    expect(dispatch.session.update).toHaveBeenCalledTimes(1)
    expect(dispatch.session.update).toHaveBeenCalledWith({ isBlogSidebarOpen: true })
  })

  it('renders the titled logo as a link that redirects to the default page', () => {
    const Link = shallow(<BlogHeader />).find('Link')
    expect(Link.prop('to')).toBe('/')
    expect(Link.find('TitledLogo')).toHaveLength(1)
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<BlogHeader />)).toMatchSnapshot()
    })
  })
})
