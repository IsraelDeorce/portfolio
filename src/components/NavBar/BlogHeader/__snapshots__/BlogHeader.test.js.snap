// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogHeader/> Snapshot renders without crashing using shallow 1`] = `
<styled.nav>
  <styled.span
    onClick={[Function]}
  >
    <MdMenu
      size="32"
    />
  </styled.span>
  <Link
    to="/"
  >
    <TitledLogo />
  </Link>
</styled.nav>
`;
